### DISCUSSION TITLE
problem with example streams-generator-a2dp

### DISCUSSION THREAD

**Discussion by Jeff-Jordan** [pschatzmann/ESP32-A2DP#345] [Q&A] [answered]
1. @Phil: AWESOME work your libraries!!!

2. Hello community,
I tried to send I2S audio data, via bluetooth, to my audio speaker and used the streams-generator-a2dp as example.
It compiles without any error and connects to my bluetooth speaker (get the connect sound from the speaker and at the serial window,  the Connecting -> Connected confirmation).

But the BT-speaker remains deaf.

Tried different speakers and different sinewaves ( sineWave.setFrequency(440); or sineWave.setFrequency(N_B4);) and messed around with the volume settings.... but always the same problem.

Board is a 30pin ESP-WROOM-32, with 4MB flash.

Kind regards and best wishes for 2023,
Jeff

**Comment by Sirlsliang** [comment]
你的邮件我已收到。

**Comment by Jeff-Jordan** [comment]
I'm still wondering why the "streams generator a2dp" example from phils audio-tools library doesn't work (connects to the BT speaker, but remains silent), but the "bt-music-sender" from his "ESP32-A2DP" library works fine.

**Reply by mcbpalisade** [reply]
> I'm still wondering why the "streams generator a2dp" example from phils audio-tools library doesn't work (connects to the BT speaker, but remains silent), but the "bt-music-sender" from his "ESP32-A2DP" library works fine.

I could not get streams to work either.  Since my project is just an I2S to BT aviation headset project thought this would be the most efficient way.  Other ways seem to have lots of noise / distortion.

**Comment by pschatzmann** [comment, marked as answer]
For me this is example is working w/o any issues. However I recommend to use the examples from the basic-api folder which are nacg more reliable.

You should try the most simple scenario first: which is the one using a generator just to make sure that you manage to get the bluetooth connection working.

If it is not working the standard ESP32 logger is your friend!
